homeassistant/components/motionblinds_ble/__init__.py

```diff
@@ -0,0 +1,104 @@
+"""Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from functools import partial
+import logging
+
+from motionblindsble.const import MotionBlindType
+from motionblindsble.crypt import MotionCrypt
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.bluetooth import (
+    BluetoothCallbackMatcher,
+    BluetoothChange,
+    BluetoothScanningMode,
+    BluetoothServiceInfoBleak,
+    async_ble_device_from_address,
+    async_register_callback,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_BLIND_TYPE, CONF_MAC_CODE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [
+    Platform.COVER,
+]
+
+CONFIG_SCHEMA = cv.empty_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Motionblinds BLE integration."""
+
+    _LOGGER.debug("Setting up Motionblinds BLE integration")
+
+    # The correct time is needed for encryption
+    _LOGGER.debug("Setting timezone for encryption: %s", hass.config.time_zone)
+    MotionCrypt.set_timezone(hass.config.time_zone)
+
+    return True
```

Comment:
Why do we need this setup here anyway?

Reply:
Is there a better way to do this? When the integration starts you need to set the timezone which is required for all commands sent to the motor

Reply:
Can't we include this in the async setup entry?

Reply:
Then you would call set_timezone for each entry which is a bit unnecessary